File: examples/wrapper-types/tests/bindings/test_wrapper_types.py
from wrapper_types import *

val = get_wrapped_types_demo(None)
assert(val.json == '{"demo":"string"}')
assert(val.handle == 123)

val = WrappedTypesDemo('{"foo":"bar"}', 456)
val2 = get_wrapped_types_demo(val)
assert(val == val2)

assert(get_next_handle(val.handle) == 457)
assert(objectify("outer", '{"hello":"there"}') == '{"outer":{"hello":"there"}}')

File: examples/wrapper-types/tests/bindings/test_wrapper_types.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import wrapper_types

// TODO: use an actual test runner.

do {
    // Test simple values.
    let demo = getWrappedTypesDemo(demo: nil)
    assert(demo.json == "{\"demo\":\"string\"}")
    assert(demo.handle == 123)
}
